src/providers/WorkflowCore.Persistence.EntityFramework/Services/EntityFrameworkPersistenceProvider.cs

```diff
@@ -185,9 +187,10 @@ public async Task PersistWorkflow(WorkflowInstance workflow, List<EventSubscript
 
                 foreach (var subscription in subscriptions)
                 {
-                    subscription.Id = Guid.NewGuid().ToString();
+                    subscription.Id = Guid.Empty.ToString();
                     var subscriptionPersistable = subscription.ToPersistable();
-                    db.Set<PersistedSubscription>().Add(subscriptionPersistable);
+                    var result = db.Set<PersistedSubscription>().Add(subscriptionPersistable);
+                    subscription.Id = result.Entity.SubscriptionId.ToString();
```

Comment:
@viktorshevchenko210 Is the Id going to be non-empty here? I thought it would get a real Id after `SaveChangesAsync` call.